system-addon/content-src/components/Search/Search.jsx

```diff
@@ -0,0 +1,34 @@
+"use strict";
+const React = require("react");
+const {connect} = require("react-redux");
+const {actionTypes, actionCreators} = require("common/Actions.jsm");
+
+const Search = React.createClass({
+  getInitialState() {
+    return {
+      searchString: ''
+    };
+  },
+  performSearch(options) {
+    let searchData = {
+      engineName: options.engineName,
+      searchString: options.searchString,
+      searchPurpose: "d",
+      healthReportKey: "newtab"
+    };
+    this.props.dispatch(actionCreators.SendToMain({type: actionTypes.PERFORM_SEARCH, data: searchData}));
+  },
+  render() {
+    const {currentEngine, engines} = this.props.Search;
+    return (<form className="search-wrapper">
+      <span className="search-label" />
+      <input value={this.state.searchString} type="search"
+        onChange={e => this.setState({searchString: e.target.value})}
+        maxLength="256"
+        placeholder="Search the Web"/>
+        <button onClick={e => {e.preventDefault(); this.performSearch({engineName: currentEngine.name, searchString: this.state.searchString});}} />
```

Comment:
I thought we turned on `eslint` `jsx-no-bind` but I guess not. The `onChange` and `onClick` here are creating new functions each time `render` is called, so this will prevent any useful caching of any previously rendered `Search`. These should be able to be plain methods on the class to avoid new functions from `render`.